vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_wallet.cc

```diff
@@ -71,6 +71,11 @@ void UpholdWallet::Generate(ledger::ResultCallback callback) const {
     return callback(type::Result::LEDGER_ERROR);
   }
 
+  if (status == type::WalletStatus::DISCONNECTED_VERIFIED) {
+    return ledger_->promotion()->TransferTokens(
+        std::bind(&UpholdWallet::OnTransferTokens, this, _1, _2, callback));
+  }
```

Comment:
Tricky — had to double check why this works, as we don't have the BAT card ID in `DISCONNECTED_VERIFIED`.

Reply:
Could we add a comment above this line with a short explanation of why we start the transfer here? I think the bitflyer and gemini cases are more obvious, so they probably don't need comments.

Reply:
Added a comment in [d020d05](https://github.com/brave/brave-core/pull/13037/commits/d020d0557e6df38ab2d0b5659bfb6be74bf21164)